pkg/reconciler/buildrun/resources/runtime_image.go

```diff
@@ -197,29 +204,70 @@ func runtimeBuildAndPushStep(b *buildv1alpha1.Build, kanikoImage string) *v1beta
 		Args: []string{
 			"--skip-tls-verify=true",
 			fmt.Sprintf("--dockerfile=%s", runtimeDockerfile),
-			fmt.Sprintf("--destination=$(params.%s%s)", prefixParamsResults, paramOutputImage),
+			fmt.Sprintf("--destination=$(params.%s%s)", prefixParamsResultsVolumes, paramOutputImage),
 			"--snapshotMode=redo",
-			"--oci-layout-path=/workspace/output/image",
+			"--oci-layout-path=/kaniko/shp-runtime-image-layout",
+		},
+		VolumeMounts: []v1.VolumeMount{
```

Comment:
Yes, workspaces have two disadvantages:

1. They are automatically mounted to all containers. But, I only need this one for the two containers. In the others (those from the build strategy), they could theoretically lead to issues. Selectively mounting workspaces only to some containers/steps of a Task is a Tekton 0.24.0 feature.
2. They need to be configured on the TaskSpec (workspace) and TaskRunSpec (workspace binding assuming one does not want to rely on the default binding). With volumes, I can configure everything on the TaskSpec which makes sense in this scenario (or actually in most if not all scenarios that we have in shipwright because we nowhere need to pass around workspaces between tasks because we do not use pipelines).